go/client/logclient.go

```diff
@@ -105,21 +115,21 @@ type getEntryAndProofResponse struct {
 	AuditPath []string `json:"audit_path"` // the corresponding proof
 }
 
-// New constructs a new LogClient instance.
+// New constructs a new logClientImpl instance.
 // |uri| is the base URI of the CT log instance to interact with, e.g.
 // http://ct.googleapis.com/pilot
 // |hc| is the underlying client to be used for HTTP requests to the CT log.
-func New(uri string, hc *http.Client) *LogClient {
+func New(uri string, hc *http.Client) LogClient {
 	if hc == nil {
 		hc = new(http.Client)
```

Comment:
Unrelated question: I've never been clear on the difference between `new(http.Client)` and `&http.Client{}`, do you know something I don't?